### PR TITLE
build the gallery on PRs

### PR DESCRIPTION
**Pull request by xster**
The script already deploys only on committed tagged dev branch commits.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @gspencergoog

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by xster** [comment]
@fkorotkov any ideas on this failure message? 

```
Gem::FilePermissionError: You don't have write permissions for the
/var/folders/xb/469gd9j941bc7v3mltn7zgb40000gn/T/bundler20181024-1269-14soujqdotenv-2.4.0/bin
```

**Comment by fkorotkov** [comment]
@xster not sure what happened there. There were no changes to macOS images in the past few weeks. Was it working before?

**Comment by gspencergoog** [comment]
@xster, did you still want to commit this?

**Comment by xster** [comment]
I do but there's some sort of authentication hang-up that isn't being properly passed by env variables I need to debug still.

File: examples/flutter_gallery/ios/fastlane/Fastfile [hunk 1]
```diff
@@ -46,14 +46,15 @@ platform :ios do
 
     puts 'Retrieving signing certificates and profiles...'
     # Stop fastlane from echoing back PUBLISHING_MATCH_CERTIFICATE_REPO var.
-    suppress_output {
+    # suppress_output {
       # Retrieves all the necessary certs and provisioning profiles.
       sync_code_signing(
-        git_url: ENV['PUBLISHING_MATCH_CERTIFICATE_REPO'],
+        git_url:  "https://x-access-token:#{ENV['CIRRUS_REPO_CLONE_TOKEN']}@#{ENV['PUBLISHING_MATCH_CERTIFICATE_REPO']}" ,
```
Comment (xster) [inline examples/flutter_gallery/ios/fastlane/Fastfile]: hey @fkorotkov, sorry to bug you again. We need to make use of git in this ruby script again with the same user session as used to clone the repo itself. This env doesn't seem to be populated. Is this the right env to look at or should we be using git some other way to authenticate?

Comment (fkorotkov) [reply examples/flutter_gallery/ios/fastlane/Fastfile]: @xster I think this part of documentation can help to answer your question: https://cirrus-ci.org/guide/writing-tasks/#custom-clone-command

Comment (xster) [reply examples/flutter_gallery/ios/fastlane/Fastfile]: That's where I got this env variable name from. Though it seems unpopulated in this PR run

Comment (fkorotkov) [reply examples/flutter_gallery/ios/fastlane/Fastfile]: It should be. By default `clone` is performed via `go-git` and not `git` executable. Not sure if it interferes with your script.

Comment (xster) [reply examples/flutter_gallery/ios/fastlane/Fastfile]: Thanks. I tried printing out various env variables listed in https://cirrus-ci.org/guide/writing-tasks/#environment-variables but only CIRRUS_REPO_CLONE_TOKEN was specifically not populated. 

Doesn't really matter too much. We'll just use our own value via encrypted variables instead. Thanks!

**Comment by xster** [comment]
Ok, it works. Submitting on green. 

This PR:

- Lets Android and iOS build the gallery on each PR
- Lets iOS try to sign it on each merge. 
- Pass in our own credential to get the certificates via https
- Renew certificates which expired
- Make a new Apple ID user to get around the 2FA craziness

**Comment by dnfield** [comment]
I think this will cause problems for anyone who isn't already a contributor - it depends on encrypted variables they can't use.

It should probably be scoped to only contributors.

**Comment by dnfield** [comment]
Ahhh I see - it's got logic in the shell script.  Nevermind.

For some reason merging this into one of my older PRs seems to have caused problems, but I think that's something else.

**Pull request merged**